File: tools/bench/answers/python_sonnet/s1/13_describe.py
def describe(v):
    if isinstance(v, bool):
        return "int " + str(int(v) + 1)
    elif isinstance(v, int):
        return "int " + str(v + 1)
    elif isinstance(v, float):
        return "float " + str(v)
    elif isinstance(v, str):
        return "str " + str(len(v))

print(describe(42))
print(describe(3.5))
print(describe("abc"))

File: tools/bench/answers/rust_sonnet/s1/13_describe.rs
enum Val {
    Int(i64),
    Float(f64),
    Str(String),
}

fn describe(v: Val) -> String {
    match v {
        Val::Int(n) => format!("int {}", n + 1),
        Val::Float(f) => format!("float {}", f),
        Val::Str(s) => format!("str {}", s.len()),
    }
}

fn main() {
    println!("{}", describe(Val::Int(42)));
    println!("{}", describe(Val::Float(3.5)));
    println!("{}", describe(Val::Str("abc".to_string())));
}
